packages/internal-test-utils/ReactInternalTestUtils.js

```diff
@@ -264,3 +264,40 @@ ${diff(expectedLog, actualLog)}
   Error.captureStackTrace(error, assertLog);
   throw error;
 }
+
+// Simulates dispatching events, waiting for microtasks in between.
+// This matches the browser behavior, which will flush microtasks
+// between each event handler.
+export async function dispatchAndWaitForDiscrete(
+  node: Node,
+  eventType: string,
+): Promise<void> {
+  assertYieldsWereCleared(waitForDiscrete);
+
+  // Bubbling phase
+  // Walk the path from the element to the document and dispatch
+  // the event on each node, flushing microtasks in between.
+  for (let current = node; current; current = current.parentNode) {
+    const customEvent = new Event(eventType, {
+      bubbles: false,
+      cancelable: true,
+    });
+
+    Object.defineProperty(customEvent, 'eventPhase', {
+      // Avoid going through the capture/bubbling phases,
+      // since we're doing it manually.
+      value: Event.AT_TARGET,
```

Comment:
Why do you need to set this if you're dispatching to each target without bubbling anyway? Shouldn't this always be `AT_TARGET` anyway?

Ideally it should only be `AT_TARGET` when `node === current` and otherwise the phase. You could potentially set it to `CAPTURING_PHASE` and add an event listener in the end of the capturing phase and then in that listener switch to `BUBBLING_PHASE`.

Not sure this matters though.

The phases will be wrong anyway since it'll be capture/bubble at each node instead of all capture and then all bubbles.